README.md

```diff
@@ -4,49 +4,250 @@
+```
+
+To define a computed property with more complex dependency expressions for
+which you may want to receive change values as arguments (e.g. sub-properties,
+splices, wildcards, etc.), or to set additional property options, define a
+standard property and set its `computed` option.
+
+```ts
+@property({computed: 'computeBaz(foo.*)', reflectToAttribute: true})
+baz: string;
+
+private computeBaz(fooChangeRecord: object) {
+  ...
+}
+```
+
+### `@observe`
+
+Define a [complex property
+observer](https://www.polymer-project.org/2.0/docs/devguide/observers#complex-observers).
+
+`targets` can be a single dependency expression, or an array of them. All
+observer dependency syntaxes are supported (property names, sub-properties,
+splices, wildcards, etc.).
+
+```ts
+@observe('foo')
+private fooChanged(newFoo: string) {
+  console.log(`foo is now: ${newFoo}`);
+}
+
+@observe(['foo', 'bar'])
+private fooBarChanged(newFoo: string, newBar: string) {
+  console.log(`foo is now: ${newFoo}, bar is now: ${newBar}`);
+}
+
+@observe('baz.*')
+private bazChanged(changeRecord: object) {
+  console.log('baz changed deeply');
+}
+```
+
+To define a [simple property
+observer](https://www.polymer-project.org/2.0/docs/devguide/observers#simple-observers),
+which receives both the old and new values, set the `observer` option on the
+property you want to observe to the observer name or (preferably) function
+reference.
+
+```ts
+@property({observer: MyElement.prototype.bazChanged})
+baz: string;
+
+private bazChanged(oldValue: string, newValue: string) {
+  console.log(`baz was: ${oldValue}, and is now: ${newValue}`);
+}
+```
+
+### `@query`
+
+Define a getter that calls
+[`querySelector`](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector)
+on the shadow root with the given `selector`.
+
+```ts
+@query('my-widget')
+widget: MyWidgetElement;
+```
+
+### `@queryAll`
+
+Define a getter that calls
+[`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelectorAll)
+on the shadow root with the given `selector`.
+
+```ts
+@queryAll('my-widget')
+widgets: NodeListOf<MyWidgetElement>
+```
+
+### `@listen`
+
+Add an event listener for `eventName` on `target`. `target` can be an object
+reference, or the string id of an element in the shadow root.
+
+Note that a `target` referenced by id must be defined statically in the
+top-level element template (e.g. not in a `<dom-if>`), because the `$` id map
+is used to find the target upon `ready()`.
+
+To use `@listen`, your element must apply the
+[`DeclarativeEventListeners`](https://github.com/Polymer/polymer-decorators/blob/master/mixins/declarative-event-listeners.html)
+mixin, which is supplied with this package.
+
+```ts
+/// <reference path="./bower_components/polymer-decorators/mixins/declarative-event-listeners.d.ts" />
+
+class MyElement extends Polymer.DeclarativeEventListeners(Polymer.Element) {
+
+  @listen('scroll', document)
+  private onDocumentScroll(ev: Event) {
```

Comment:
I think `private` generally causes a warning that the method isn't called anywhere. Use `protected` to get around that.